Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -4623,8 +4623,9 @@ void WebPageProxy::continueNavigationInNewProcess(API::Navigation& navigation, W
     }
 
     RegistrableDomain navigationDomain { navigation.currentRequest().url() };
-    Ref frameProcess = m_browsingContextGroup->ensureProcessForDomain(navigationDomain, newProcess, preferences());
-    m_provisionalPage = makeUnique<ProvisionalPageProxy>(*this, WTFMove(frameProcess), m_browsingContextGroup, WTFMove(suspendedPage), navigation, isServerSideRedirect, navigation.currentRequest(), processSwapRequestedByClient, isProcessSwappingOnNavigationResponse, websitePolicies.get(), replacedDataStoreForWebArchiveLoad);
+    Ref browsingContextGroup = newProcess->websiteDataStore() == &websiteDataStore() ? m_browsingContextGroup : BrowsingContextGroup::create();
```

Comment:
My first thought was that this should be based on looking at the WebsitePolicies, if its WebsiteDataStore is set and not equal to the opening page's data store.  This is likely equivalent, though.